apps/storefront/src/app/[locale]/(main)/products/[slug]/components/product-details-container.tsx

```diff
@@ -0,0 +1,75 @@
+import { cookies } from "next/headers";
+import { notFound } from "next/navigation";
+
+import { getAccessToken } from "@/auth";
+import { CACHE_TTL, COOKIE_KEY } from "@/config";
+import { clientEnvs } from "@/envs/client";
+import { JsonLd, productToJsonLd } from "@/lib/json-ld";
+import { getCurrentRegion } from "@/regions/server";
+import { cartService, storeService, userService } from "@/services";
+import { storefrontLogger } from "@/services/logging";
+
+import { ProductDetails } from "./product-details";
+
+export const ProductDetailsContainer = async ({
+  params,
+}: {
+  params: Promise<{ slug: string }>;
+}) => {
+  const { slug } = await params;
+
+  const [region, accessToken, checkoutId] = await Promise.all([
+    getCurrentRegion(),
+    getAccessToken(),
+    (async () => (await cookies()).get(COOKIE_KEY.checkoutId)?.value)(),
```

Comment:
Does this does not work?
```suggestion
    cookies()).get(COOKIE_KEY.checkoutId)?.value
```

Reply:
Nope, `cookies().get(...)` returns a cookie object, not a promise.